src/main/java/entities/Post.java

```diff
@@ -12,15 +13,34 @@
     private User owner;
     private Recipe recipe;
     private String category;
+    private int idPost;
     private LocalDateTime postedTime;
 
-    public Post(User owner, LocalDateTime postedTime, Recipe recipe, String category) {
+    public Post(User owner, LocalDateTime postedTime, Recipe recipe, String category, int pId) {
         this.likedUsers = new ArrayList<>();
         this.comments = new ArrayList<>();
         this.owner = owner;
         this.recipe = recipe;
         this.category = category;
         this.postedTime = postedTime;
+        this.idPost = pId;
+    }
+
+    public int getIdPost() {
+        return this.idPost;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I think we could assume that the param is of type Post. It would never make sense, I think, to compare a Post with a different type of object for equality.

Reply:
I think we can leave it just in case, doesn't hurt right?

Reply:
I don't feel that we should ever feel that we should be comparing a Post with something that isn't a Post. Can you think of a scenario where we would?

The reason is that this method could be much simpler (1 line)

equals (Post post) {
return this.postId == post.getPostId();
}